bodo/pandas/physical/reduce.h

```diff
@@ -48,36 +48,40 @@ class PhysicalReduce : public PhysicalSource, public PhysicalSink {
             false /*convert_timedelta_to_int64*/, "", time_unit,
             false, /*downcast_time_ns_to_us*/
             bodo::default_buffer_memory_manager());
-
-        // Reduce Arrow array using compute function
-        arrow::Result<arrow::Datum> cmp_res =
-            arrow::compute::CallFunction(function_name, {in_arrow_array});
-        CHECK_ARROW(cmp_res.status(), "Error in Arrow compute kernel");
-        std::shared_ptr<arrow::Scalar> out_scalar_batch =
-            cmp_res.ValueOrDie().scalar();
-
-        ReductionType reduction_type = getReductionType(function_name);
-
-        // Update reduction result
-        if (iter == 0) {
-            output_scalar = out_scalar_batch;
-        } else {
-            arrow::Result<arrow::Datum> cmp_res_scalar =
-                arrow::compute::CallFunction(scalar_cmp_name,
-                                             {out_scalar_batch, output_scalar});
-            CHECK_ARROW(cmp_res_scalar.status(),
-                        "Error in Arrow compute scalar comparison");
-            const std::shared_ptr<arrow::Scalar> cmp_scalar =
-                cmp_res_scalar.ValueOrDie().scalar();
-            if (reduction_type == ReductionType::COMPARISON) {
-                if (cmp_scalar->Equals(arrow::BooleanScalar(true))) {
-                    output_scalar = out_scalar_batch;
-                }
-            } else if (reduction_type == ReductionType::AGGREGATION) {
-                output_scalar = cmp_scalar;
+        std::vector<std::shared_ptr<arrow::Scalar>> output_scalar_batches;
+        for (size_t i = 0; i < function_names.size(); i++) {
+            auto function_name = function_names[i];
+            // Reduce Arrow array using compute function
+            arrow::Result<arrow::Datum> cmp_res =
+                arrow::compute::CallFunction(function_name, {in_arrow_array});
+            CHECK_ARROW(cmp_res.status(), "Error in Arrow compute kernel");
+            std::shared_ptr<arrow::Scalar> out_scalar_batch =
+                cmp_res.ValueOrDie().scalar();
+            output_scalar_batches.push_back(out_scalar_batch);
+            ReductionType reduction_type = getReductionType(function_name);
+
+            // Update reduction result
+            if (iter == 0) {
+                output_scalars = output_scalar_batches;
```

Comment:
I think it would be a bit simpler to push back here instead:
```suggestion
                output_scalars.push_back(out_scalar_batch);
```